domains/service/src/system_domain.rs

```diff
@@ -128,12 +133,51 @@ where
         task_manager,
         transaction_pool,
         select_chain: (),
-        other: (telemetry, telemetry_worker_handle, executor),
+        other: (
+            telemetry,
+            telemetry_worker_handle,
+            executor,
+            import_block_notification_stream,
+        ),
     };
 
     Ok(params)
 }
 
+/// Secondary chain configuration.
+pub struct Configuration {
```

Comment:
Unsure this abstraction is necessary, you can just add another argument to the function `new_full` which is much simpler.

Reply:
Will be complex down the line as we add more fields. Hence abstracted it out so we can add without too much breakage